pkg/driver/controller_server.go

```diff
@@ -457,6 +455,32 @@ func (driver *Driver) createVolume(
 				return nil, status.Error(codes.Internal, "Parent volume ID is missing in the snapshot response")
 			}
 
+			// Get the parent volume from the snapshot
+			existingParentVolume, err := storageProvider.GetVolume(existingSnap.VolumeID)
+			if err != nil {
+				log.Error("err: ", err.Error())
+				return nil,
+					status.Error(codes.Internal,
+						fmt.Sprintf("Failed to check if snapshot's parent volume %s with ID %s exist, err: %s",
+							existingSnap.VolumeName, existingSnap.VolumeID, err.Error()))
+			}
+
```

Comment:
As @shivamerla pointed, 'GetVolume' is expected to return both 'existingParentVolume' and 'err' as nil when volume is not found. You have already handled err != nil case, but still need to handle the other case when 'existingParentVolume' is nil.

Please add the below check: 
if existingParentVolume == nil {
	return nil, status.Error(codes.NotFound, "Could not find parent volume with ID "+volID)
}

Reply:
Ok, thanks. I will handle it although this is not my code change and it has probably come from upstream after merge. 
Here is the relevant PR . 
https://github.com/hpe-storage/csi-driver/pull/108/commits/7c9aebc84d892a01bce77be363ee24a9ba0ad372